pkg/server/server.go

```diff
@@ -456,6 +456,12 @@ func (s *Server) Run(ctx context.Context) error {
 		}
 	}
 
+	if s.Options.Cache.Enabled && (s.Options.Controllers.EnableAll || enabled.Has("apiexportendpointslice")) {
```

Comment:
So this doesn't work if the cache server is disabled? What is the fallback option for a service provider wanting to know which URLs to use?

Reply:
Yes. I should have raised that. Here are my views:
- sharded environment means cache server
- the construct (APIExportEndpointSlice) is of little value for non-sharded environment
- it is not a good UX for service providers to have a different API for sharded and non-sharded environments. Most probably they want to develop their service once and have it working for both

I would propose that in a non-sharded / no cache server environment informers for the local instance are passed to the controller instead of the ones for the cache server, e.g.:
~~~
c, err := apiexportendpointslice.NewController(
		s.KcpSharedInformerFactory.Apis().V1alpha1().APIExportEndpointSlices(),
		// ClusterWorkspaceShards and APIExports get retrieved from cache server
		s.CacheKcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaceShards(),
		s.CacheKcpSharedInformerFactory.Apis().V1alpha1().APIExports(),
		kcpClusterClient,
	)
~~~
vs
~~~
c, err := apiexportendpointslice.NewController(
		s.KcpSharedInformerFactory.Apis().V1alpha1().APIExportEndpointSlices(),
		// ClusterWorkspaceShards and APIExports get retrieved from cache server
		s.KcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaceShards(),
		s.KcpSharedInformerFactory.Apis().V1alpha1().APIExports(),
		kcpClusterClient,
	)
~~~
 As the cache server is K8 API compliant the same code should work with both. As there is a single shard all the relevant information should be provided by it.
 
 What do you think?

Reply:
We usually pass in both and have the controller try the local ones first, then fall back to the cache server ones. Right @stevekuznetsov @p0lyn0mial @sttts ?

Reply:
Let's not diverge between local and sharded mode. Let's run the cache server even in single-shard mode.

Reply:
As discussed I will modify the flags so that the cache server is started in embedded mode by default and as a separate process if additional information is provided.